core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c

```diff
@@ -963,19 +968,20 @@ static unsigned int clk_freq_prop(void *fdt, int node)
 
 static void get_osc_freq_from_dt(void *fdt)
 {
-	enum stm32mp_osc_id idx;
+	enum stm32mp_osc_id idx = _UNKNOWN_OSC_ID;
 	int clk_node = fdt_path_offset(fdt, "/clocks");
 
 	if (clk_node < 0)
 		panic();
 
-	for (idx = (enum stm32mp_osc_id)0 ; idx < NB_OSC; idx++) {
+	assert((int)_HSI == 0);
```

Comment:
Why is this needed? Consider using `COMPILE_TIME_ASSERT()` if it's needed at all.

Reply:
True, thanks!
I'll use `COMPILE_TIME_ASSERT()`. I think it is preferable for readers to understand that this loop expects to cover all the relevant IDs of this enumerated type.